src/TransitionWrapper.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useCallback, useRef } from 'react';
+import Transition, {
+  TransitionProps,
+  TransitionStatus,
+} from 'react-transition-group/Transition';
+import safeFindDOMNode from 'react-overlays/safeFindDOMNode';
+import useMergedRefs from '@restart/hooks/useMergedRefs';
+
+export type TransitionWrapperProps = TransitionProps & {
+  childRef?: React.Ref<unknown>;
+  children:
+    | React.ReactElement
+    | ((
+        status: TransitionStatus,
+        props: Record<string, unknown>,
+      ) => React.ReactNode);
+};
+
+// Normalizes Transition callbacks when nodeRef is used.
+const TransitionWrapper = React.forwardRef<
```

Comment:
I can never remember what RTG is doing now...do we need this wrapper? If it's in v5 can we just break the API and require node ref compatible components?

Reply:
RTG has this quirk where the node is excluded in the callback signature when using nodeRef.

ie.  `(node, isAppearing) => ...` becomes `(isAppearing) => ...`

This wrapper abstracts all that so callers don't have to worry about it (like they can just use it like it was without using nodeRef).  I kinda figured this would be a cleaner approach.

Reply:
ah, that seems like a bad API choice i probably ok'd oops